cmd/crc/cmd/daemon_nonwin.go

```diff
@@ -13,7 +13,7 @@ import (
 func runDaemon() {
 	// Remove if an old socket is present
 	os.Remove(constants.DaemonSocketPath)
-	crcAPIServer, err := api.CreateAPIServer(constants.DaemonSocketPath, newConfig, newMachineWithConfig)
+	crcAPIServer, err := api.CreateAPIServer(constants.DaemonSocketPath, config, newMachine())
```

Comment:
Does this impact thread safety? The current code is already a bit scary imo, and I'm worried this adds more shared state between threads?

Reply:
config is not thread safe at all. Not better before or after this commit. I plan to submit this after: https://github.com/code-ready/crc/compare/master...guillaumerose:syncmachine?expand=1 which helps a bit.

At least now, one config and only one machineClient are created.

Reply:
> At least now, one config and only one machineClient are created.

This is precisely my concern ;) Before, we were constantly recreating the config object, so each thread had its own config instance that no other thread would try to access. If all threads are now sharing the same object, this commit is making things worse.

Reply:
This is now better for config. For machine, it will come right after.